test/88mph/88mph-uni.js

```diff
@@ -0,0 +1,362 @@
+  let priceOracle;
+
+  let feeForwarder;
+  let unidex = "0xde2d1a51640f78257713031680d1f306297d957426e912ab21317b9cc9495a41";
+
+  const percentageFeeModelContract = "0x9c2ae492ec3A49c769bABffC9500256749404f8E";
+
+  // parties in the protocol
+  let governance;
+  let farmer1;
+  let farmer2;
+
+  // numbers used in tests
+  let farmerBalance;
+
+  // Core protocol contracts
+  let controller;
+  let vault;
+  let strategy;
+  let potPool;
+
+  let farmerInitialBalance;
+
+  async function setupExternalContracts() {
+    underlying = await IERC20.at("0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984");
+    console.log("Fetching Underlying at: ", underlying.address);
+    priceOracle = await IOracleMainnet.at(priceOracleAddr);
+  }
+
+  async function setupBalance(){
+    let etherGiver = accounts[9];
+    // Give whale some ether to make sure the following actions are good
+    await web3.eth.sendTransaction({ from: etherGiver, to: underlyingWhale, value: 10e18});
+    await web3.eth.sendTransaction({ from: etherGiver, to: underlyingWhale2, value: 10e18});
+    await web3.eth.sendTransaction({ from: etherGiver, to: governance, value: 10e18});
+
+    farmerBalance = await underlying.balanceOf(underlyingWhale);
+    await underlying.transfer(farmer1, farmerBalance, { from: underlyingWhale });
+    const farmer2Balance = await underlying.balanceOf(underlyingWhale2);
+    await underlying.transfer(farmer2, farmer2Balance, { from: underlyingWhale2 });
+  }
+
+  before(async () => {
+    governance = "0xf00dD244228F51547f0563e60bCa65a30FBF5f7f";
+    accounts = await web3.eth.getAccounts();
+
+    farmer1 = accounts[1];
+    farmer2 = accounts[2];
+
+    // impersonate accounts
+    await impersonates([governance, underlyingWhale, underlyingWhale2]);
+
+    await setupExternalContracts();
+
+    // whale send underlying to farmers
+    await setupBalance();
+
+    [controller, vault, strategy, potPool] = await setupCoreProtocol({
+      "existingVaultAddress": null,
+      "strategyArtifact": Strategy,
+      "strategyArtifactIsUpgradable": true,
+      "underlying": underlying,
+      "governance": governance,
+      "liquidation": [
+        {"sushi": [mph, weth]}, 
+        {"uniV3": [weth, underlying.address]}, 
+        {"uniV3": [underlying.address, weth]}
+      ],
+      "rewardPool" : true,
+      "rewardPoolConfig": {
+        type: 'PotPool',
+        rewardTokens: [
+          addresses.IFARM,
+          xMph
+        ]
+      },
+    });
+
+    await strategy.setPotPool(potPool.address, {from: governance});
+    await potPool.setRewardDistribution([strategy.address], true, {from: governance});
+
+    feeForwarder = await IFeeRewardForwarder.at(await controller.feeRewardForwarder());
+    await feeForwarder.configureLiquidation(
+      [weth, addresses.FARM],
+      [unidex],
+      {from: governance}
+    );
+  });
+
+  describe("Happy path", () => {
+    it("Farmer should earn money... and more!", async () => {
```

Comment:
I added some extra steps to this test such as
 - simulate waiving the early withdrawal fee for our deposit (done by 88mph)
 - testing deposits, top-ups, rollover etc.
 
 Have a look at the test output in the description of this PR